core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,22 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    @DataBoundSetter
+    public void setViews(Collection<View> views) throws IOException {
```

Comment:
I doubt synchronization is relevant here, since this would only be called from inside a system configuration script, probably during startup. But perhaps this is intended to resynch a Jenkins instance which is already up and running? I do not know.